zookeeper-server/src/main/java/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -574,6 +585,41 @@ public void processRequest(Request request) {
         }
     }
 
+    private Record handleReadRequest(int opCode, Record request, ServerCnxn cnxn, List<Id> authInfo)
+            throws KeeperException, IOException {
+        switch(opCode) {
+            case OpCode.getChildren: {
+                GetChildrenRequest getChildrenRequest = (GetChildrenRequest) request;
```

Comment:
Please refactor the code to reuse existing logic for getChildren and getData, which makes it easier to maintain in the future.

Reply:
Hmm, that is something that I 100% percent agree but I do not see why this code is not appropriate. The code of getChildren and getData only creates the record from the request and then call the `handleGetChildrenRequest` or `handleGetDataRequest` function as well just like the multiRead version.
Could we improve upon this?

Reply:
I merged your change into on our internal FB branch for getDataList and needed to do some manual refactoring to get FinalRequestProcessor working.  As part of this I defined 3 new private methods:
- handleGetChildrenRequest
- handleGetDataRequest
- verifyPathExistsAndCheckACL: an internal helper that reduces duplicate code throughout the class

The cleanest thing about the refactoring is that the switch statement on the multiRead type only needs to be called once.  Let me upload a patch file

Reply:
nm, looks like you made the changes in line with the suggestion.  looks good.